pkgs/native_assets_cli/lib/src/model/resource_identifiers.dart

```diff
@@ -0,0 +1,202 @@
+    final listEquals = const DeepCollectionEquality().equals;
+
+    return other is ResourceIdentifiers &&
+        listEquals(other.identifiers, identifiers);
+  }
+
+  @override
+  int get hashCode => identifiers.hashCode;
+
+  @override
+  String toString() => 'ResourceIdentifiers(identifiers: $identifiers)';
+}
+
+class Identifier {
+  final String name;
+  final String id;
+  final Uri uri;
+  final bool nonConstant;
+  final List<ResourceFile> files;
+
+  Identifier({
+    required this.name,
+    required this.id,
+    required this.uri,
+    required this.nonConstant,
+    required this.files,
+  });
+
+  Map<String, dynamic> toJson() => {
+        'name': name,
+        'id': id,
+        'uri': uri.toFilePath(),
+        'nonConstant': nonConstant,
+        'files': files.map((x) => x.toJson()).toList(),
+      };
+
+  @override
+  String toString() =>
+      '''Identifier(name: $name, id: $id, uri: $uri, nonConstant: $nonConstant, files: $files)''';
+
+  factory Identifier.fromJson(Map<String, dynamic> map) => Identifier(
+        name: map['name'] as String,
+        id: map['id'] as String,
+        uri: Uri.file(map['uri'] as String),
+        nonConstant: map['nonConstant'] as bool,
+        files: List<ResourceFile>.from((map['files'] as List)
+            .map((e) => e as Map<String, dynamic>)
+            .map(ResourceFile.fromJson)),
+      );
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) return true;
+    final listEquals = const DeepCollectionEquality().equals;
+
+    return other is Identifier &&
+        other.name == name &&
+        other.id == id &&
+        other.uri == uri &&
+        other.nonConstant == nonConstant &&
+        listEquals(other.files, files);
+  }
+
+  @override
+  int get hashCode =>
+      name.hashCode ^
+      id.hashCode ^
+      uri.hashCode ^
+      nonConstant.hashCode ^
+      files.hashCode;
+}
+
+class ResourceFile {
+  final int part;
+  final List<ResourceReference> references;
+
+  ResourceFile({required this.part, required this.references});
+
+  Map<String, dynamic> toJson() => {
+        'part': part,
+        'references': references.map((x) => x.toJson()).toList(),
+      };
+
+  factory ResourceFile.fromJson(Map<String, dynamic> map) => ResourceFile(
+        part: map['part'] as int,
+        references: List<ResourceReference>.from((map['references'] as List)
+            .map((e) => e as Map<String, dynamic>)
+            .map(ResourceReference.fromJson)),
+      );
+
+  @override
+  String toString() => 'ResourceFile(part: $part, references: $references)';
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) return true;
+    final listEquals = const DeepCollectionEquality().equals;
+
+    return other is ResourceFile &&
+        other.part == part &&
+        listEquals(other.references, references);
+  }
+
+  @override
+  int get hashCode => part.hashCode ^ references.hashCode;
```

Comment:
If you expect these objects to ever be in a collections with other types of objects, consider xorring some random prime to your linking as well.